BraintreeUIKit/Models/BTUIKCardType.m

```diff
@@ -84,17 +84,9 @@ + (NSArray *)possibleCardTypesForNumber:(NSString *)number {
         return [[self class] allCards];
     }
     NSMutableSet *possibleCardTypes = [NSMutableSet set];
-    for (BTUIKCardType *cardType in [[self class] allCards]) {
-        for (NSString *prefix in cardType.validNumberPrefixes) {
-            NSUInteger compareLength = MIN(prefix.length, number.length);
-
-            NSString *sizedPrefix = [prefix substringToIndex:compareLength];
-            NSString *sizedNumber = [number substringToIndex:compareLength];
-            if ([sizedNumber isEqualToString:sizedPrefix]) {
-                [possibleCardTypes addObject:cardType];
-                break;
-            }
-        }
+    BTUIKCardType *card = [self cardTypeForNumber: number];
+    if (card) {
+        [possibleCardTypes addObject:card];
```

Comment:
Should this change repeat the new logic in lines 66-73 but don't do an early return? (So it should keep adding cards instead of just the first one). For the cases where this an overlap in the regular expressions.